### PR TITLE
Generated fields privatized for PHP; add getters/setters

### PR DESCRIPTION
**Pull request by 2DSharp**
Solves: https://github.com/facebook/fbthrift/issues/352

Java-style private fields added with accessors/mutators

### CODE REVIEW

File: thrift/compiler/generate/t_php_generator.cc
Comment (andrii-korotkov) [inline thrift/compiler/generate/t_php_generator.cc:1374]: Using 'get' universally is better.

Comment (2DSharp) [reply thrift/compiler/generate/t_php_generator.cc:1374]: It's been a long time since I made this PR, could you clarify what you meant?

Comment (2DSharp) [reply thrift/compiler/generate/t_php_generator.cc:1374]: Ahh, I think you meant using just `get` instead of `is`. Yes, I would agree.

File: thrift/compiler/generate/t_php_generator.cc
Comment (andrii-korotkov) [inline thrift/compiler/generate/t_php_generator.cc:1269]: This would require a migration from all users. Keep this public for some time.

Comment (2DSharp) [reply thrift/compiler/generate/t_php_generator.cc:1269]: Understood.

**Review by andrii-korotkov — CHANGES_REQUESTED** [review]
I think getters and setters with no extra semantic just add an extra complexity. 

Unions need getters/setters, since there's an extra semantic of having at least one field set. In C++, there's an extra __isset struct that needs special handling. PHP Thrift structs are essentially fancy data bags, thus a direct field access doesn't have downsides IMO.

**Review by andrii-korotkov — CHANGES_REQUESTED** [review]
Sorry for getting to this late. The issue is fixed as of now.

**Comment by 2DSharp** [comment]
> I think getters and setters with no extra semantic just add an extra complexity.

I get your point, however, as an user, I had to deal with both the Java and PHP versions of thrift, and I was confused with the inconsistent behavior in both. I think having an extra getter wouldn't hurt, since most PHP applications do follow OOP-Java styled practices anyway, whilst making it consistent.

File: thrift/compiler/generate/t_php_generator.cc
Comment (andrii-korotkov) [inline thrift/compiler/generate/t_php_generator.cc:1348]: nit: This can be simplified by
`for (const auto& field : tstruct->get_members()) { ... }`

**Review by andrii-korotkov — COMMENTED** [review]
Yeah, consistency between languages is also an important factor. Let's introduce the getters/setters and follow-up on making a field private or not later.

Also, if you want, there's `t_hack_generator.cc` which may benefit from similar changes.

**Comment by 2DSharp** [comment]
Great! I had built my own version of fbthrift with getters/setters and I think it would help someone else out too, and introducing privates may make sense in a future breaking release. I did have the hack version in mind, but I thought I would wait for approval on this one first.

File: thrift/compiler/generate/t_php_generator.cc
Comment (2DSharp) [reply thrift/compiler/generate/t_php_generator.cc]: Done!

Comment (2DSharp) [reply thrift/compiler/generate/t_php_generator.cc]: Done! Although since I am using the Github web version, I missed out on the brace and can't squash the commits from my end. If it has to be merged, could you squash it from your end?

**Comment by 2DSharp** [comment]
I have made the changes you asked for. Let me know if anything else needs to be changed. And if this PR works out fine, I will move to hack.

**Comment by 2DSharp** [comment]
Can this be merged? It has been waiting for over a year now.

**Review by andrii-korotkov — APPROVED** [review]

**Review by andrii-korotkov — APPROVED** [review]

**Review by andrii-korotkov — APPROVED** [review]

**Comment by andrii-korotkov** [comment]
I've imported it and waiting for a final review and land. Thanks for contributing!

**Comment by 2DSharp** [comment]
@andrii-korotkov I figured that the parameter "type" is not being used anymore as I have removed the `is`-prefixed functions. Should I add a commit to remove that?

**Comment by andrii-korotkov** [comment]
No, that's okay to keep

**Comment by vitaut** [comment]
The PHP generator is no longer supported.

**Pull request closed without merging**
